### PR TITLE
Perf trace cleanup, etc

### PR DESCRIPTION
**Pull request by mck1117**
- Properly disable via ENABLE_PERF_TRACE
- Move enable defines to `chconf_common.h`
- Trace global lock/unlock events
- Remove `EFI_CLOCK_LOCKS`: we now profile with the new system

### CODE REVIEW

**Comment by rusefi** [comment]
Can one of the admins verify this patch?

**Comment by rusefi** [comment]
Add to whitelist

File: firmware/config/chconf_common.h [hunk 1]
```diff
@@ -12,6 +12,12 @@
 #include "obd_error_codes.h"
 #endif /* _FROM_ASM_ */
 
+
+#ifndef ENABLE_PERF_TRACE
+// looks like this value could not be defined in efifeatures.h - please define either externally or just change the value here
+ #define ENABLE_PERF_TRACE FALSE
```
Comment (mck1117) [inline firmware/config/chconf_common.h]: @rusefi Do we have a preference about the default here? Normal users probably don't have a need to enable it, and there is a small (but extant) performance impact to leaving it on.

Comment (rusefi) [reply firmware/config/chconf_common.h]: My preference is TRUE
I would hope that with our gigantic CPU we shall afford some monitoring features

File: firmware/controllers/trigger/trigger_central.cpp [hunk 2]
```diff
@@ -509,12 +509,8 @@ extern PwmConfig triggerSignal;
 extern uint32_t hipLastExecutionCount;
 extern uint32_t hwSetTimerDuration;
 
-extern uint32_t maxLockedDuration;
```
Comment (rusefi) [inline firmware/controllers/trigger/trigger_central.cpp]: why do we remove maxLockedDuration? in my opinion this is an important and unrelated feature

**Comment by rusefi** [comment]
I am not comfortable with the size and multiple changes bundled together into one PR

this is not a perf trace cleanup since I see 1) clean-up 2) new feature 3) removal of old unrelated feature

Please close this PR and consider submitting this as a few smaller changes

**Comment by rusefi** [comment]
@mck1117 can you please help this PR?

i was hesitant to merge. Maybe we can take care of https://github.com/rusefi/rusefi_documentation/issues/28 and return to this after documenting our solution?

**Pull request closed without merging**